### PR TITLE
AVRO-3724 Fix C# JsonEncoder for nested array of records

### PR DESCRIPTION
**Pull request by rayokota**
## What is the purpose of the change

https://issues.apache.org/jira/browse/AVRO-3724

Fix C# JsonEncoder for nested array of records.

The SpecificDatumWriter is missing a call to `encoder.StartItem()`.  The GenericDatumWriter has this call.

## Verifying this change

Added test with nested array of records that fails without the fix and passes with the fix.

## Documentation

N/A

### CODE REVIEW

**Comment by rayokota** [comment]
@martin-g , @KyleSchoonover , hi again!  Could I kindly get a review?  Thanks in advance!

**Review by zcsizmadia — APPROVED** [review]

**Comment by zcsizmadia** [comment]
Lgtm

File: lang/csharp/src/apache/test/IO/JsonCodecTests.cs [hunk 1]
```diff
@@ -285,6 +286,28 @@
...
+
+            ByteBufferOutputStream bbos = new ByteBufferOutputStream();
+
+            Encoder encoder = new JsonEncoder(data.Schema, bbos);
+            writer.Write(data, encoder);
+            encoder.Flush();
+
+            List<MemoryStream> listStreams = bbos.GetBufferList();
+
+            StreamReader reader = new StreamReader(listStreams[0]);
```
[the remark this answers is not in the document]
Comment (martin-g) [reply lang/csharp/src/apache/test/IO/JsonCodecTests.cs:306]: @rayokota Should we fix this warning ?

[the remark this answers is not in the document]
Comment (zcsizmadia) [reply lang/csharp/src/apache/test/IO/JsonCodecTests.cs:306]: Yes. I was looking at the PR on my hone and could not see it.

[the remark this answers is not in the document]
Comment (zcsizmadia) [reply lang/csharp/src/apache/test/IO/JsonCodecTests.cs:306]: Just needs `using` at the beginning of that line with the warning

**Review by zcsizmadia — CHANGES_REQUESTED** [review]

File: lang/csharp/src/apache/test/IO/JsonCodecTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (rayokota) [reply lang/csharp/src/apache/test/IO/JsonCodecTests.cs]: Thanks @martin-g , @zcsizmadia , I fixed the warning by adding a using clause.

**Review by zcsizmadia — APPROVED** [review]

**Comment by martin-g** [comment]
Thank you, @rayokota !

**Comment by rayokota** [comment]
Thanks for the quick review, @martin-g , @zcsizmadia !

**Comment by andrewbenedek** [comment]
Thank you for fixing and merging this so quickly everyone, your efforts are very much appreciated.  I've pulled the latest code and can confirm that I can now produce the Avro JSON for my schema.

**Pull request merged**